rllib/core/models/catalog.py

```diff
@@ -0,0 +1,227 @@
+import gymnasium as gym
+from ray.rllib.core.models.configs import (
+    MLPEncoderConfig,
+    LSTMEncoderConfig,
+    CNNEncoderConfig,
+)
+from ray.rllib.core.models.base import ModelConfig
+from ray.rllib.models import MODEL_DEFAULTS
+from gymnasium.spaces import Box
+from ray.rllib.models.utils import get_filter_config
+
+
+class Catalog:
```

Comment:
I'm somewhat happy with that, especially since that was part of my original proposal.
But I have an issue with it. I believe that the beauty of Catalog is actually, that you can do pretty much whatever suits your purpose with it. For example, you can overwrite methods such as get_encoder_config or later something like get_preprocessor or get_action_dist. Not I see myself doing that everyday, but we'd certainly take way capacity of using this as a direct way of configuration that I think is really great.

Reply:
yep yep. I agree with you. It was a bad suggestion.